src/features/hotspots/root/WelcomeOverview.tsx

```diff
@@ -78,6 +82,54 @@ const WelcomeOverview = ({ accountRewards }: Props) => {
     (state: RootState) => state.validators.validators.loading,
   )
 
+  const accountAddress = useSelector(
+    (state: RootState) => state.account.account?.address,
+  )
+
+  const hotspotRewards = useSelector(
+    (state: RootState) => state.hotspots.rewards || {},
+  )
+
+  const currentOraclePrice = useSelector(
+    (state: RootState) => state.heliumData.currentOraclePrice,
+  )
+
+  // Hook that is used for helium balance widget.
+  useAsync(async () => {
+    const token = await getWalletApiToken()
+    const oraclePrice = currentOraclePrice?.price
+    const floatBalance = oraclePrice?.floatBalance
+
+    await SharedGroupPreferences.setItem(
+      'myBalanceWidgetKey',
+      {
+        hntPrice: floatBalance,
```

Comment:
Will the oracle price only update when you open the app then? Would we want to fetch this from the widget?

Reply:
Good catch! Will fetch this from the widget.

- [ ] Fetch

Reply:
Now fetching from the widget. This will be used for the default value